### ISSUE TITLE
--retry-interval-max didn't change the max retry interval

### ISSUE THREAD

**Issue by jakeslee** [kubernetes-csi/external-provisioner#849]
I use command line option `--retry-interval-max=1h` to start a provisioner container as sidecar with my custom CSI plugin. According to the manual(https://github.com/kubernetes-csi/external-provisioner#command-line-options) introduced, I expect the retry interval of failed provisioning is up to 1 hour, but `CreateVolume` was called every 15 minutes actually.

I found that codes in controllers.go setup a periodic resync job which produce an Add Event every 15 minutes.

https://github.com/kubernetes-csi/external-provisioner/blob/21c2c40411fa56ba321adf097057b8de3824a560/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/v8/controller/controller.go#L687-L692

So, is that designed? 

**What happened**:

call `CreateVolume` every 15 minuts.

**What you expected to happen**:

call `CreateVolume` per hour.

**How to reproduce it**:

set `--retry-interval-max=1h` and `CreateVolume` returns error continuously.

**Anything else we need to know?**: 

**Environment**:
- Driver version: csi-provisioner:v2.2.2
- Kubernetes version (use `kubectl version`): OCP 4.8
- OS (e.g. from /etc/os-release): CoreOS

**Comment by msau42** [comment]
v2.2 is out of support now. Can you upgrade to a newer version and see if it's reproducible?
/triage needs-more-information

**Comment by jakeslee** [comment]
I checked this issue in v3.4.0 and master branch. I found that the problem is caused by Informer `Resync` mechanism.

The `ProvisionController` register an event handler  `ResourceEventHandlerFuncs` to handle the `Sync` event. When `Sync` event was distributed, the function `UpdateFunc` of the handler will be called, then enqueue PVC to `claimQueue`.

https://github.com/kubernetes-csi/external-provisioner/blob/831b9c5d80374f673f16bcd615f40f5a832ed0bf/vendor/sigs.k8s.io/sig-storage-lib-external-provisioner/v8/controller/controller.go#L673-L692

`controller.enqueueClaim()` function adds the PVC to `claimQueue` processing queue directly, so that the item does not follow the exponential ratelimit(call `AddRateLimited()` followed, `Add()` didn't).

In the end, PVC dequeued in function `ProvisionController.processNextClaimWorkItem()` is used to do provision operation(call `CreateVolume`).

I wrote a nested exponential ratelimiter in CSI plugin. Is there better trick to deal with case like that?
